packages/connector/src/utils.ts

```diff
@@ -76,3 +78,15 @@ export const isSnapInstalled = async (
     return false;
   }
 };
+
+export const checkSnapSupport = async () => {
+  const version = await window?.ethereum?.request?.({
+    method: 'web3_clientVersion',
+  });
+
+  const currentVersion = version.match(/\/v?(\d+\.\d+\.\d+)/u)?.[1] ?? null;
+  const isMobile = version.endsWith('Mobile');
+  return (
+    compare(METAMASK_SUPPORTS_SNAPS_VERSION, currentVersion, '<=') && !isMobile
```

Comment:
why vice versa?

IMO `compare(currentVersion, METAMASK_SUPPORTS_SNAPS_VERSION, '>=')` is better